simulators/common.py

```diff
@@ -2,36 +2,54 @@
 
 
 class BaseSystem(object):
+    """This class, as its name says, it is the `System` class from which every
+    other `System` class is inherited. If a custom command that can be useful
+    for every kind of simulator has to be implemented, this class is the right
+    place."""
 
     __metaclass__ = abc.ABCMeta
 
     @staticmethod
     def system_stop():
-        """This method sends back to the server the message `$server_shutdown!`
+        """This method sends back to the server the message `$server_shutdown%`
         ordering it to stop accepting requests and to shut down."""
-        return '$server_shutdown!'
+        return '$server_shutdown%'
 
 
 class ListeningSystem(BaseSystem):
+    """This class implements a server that wait for its client(s) to send a
+    command, it can then answer back when required. Unlike the `SendingSystem`
+    described below, it does not start any communication with any client right
+    after the connection is established."""
 
     @abc.abstractmethod
     def parse(self, byte):
-        """This method takes a byte (single character string) and returns:
-        False when the given byte is not the header, but the header is
-        expected, True when the given byte is the header or a following
-        expected byte, the response (the string to be sent back to the client)
-        when the message is completed.
-        The method eventually raises a ValueError in one of the following
-        cases: the declared length of the message exceeds the maximum expected
-        length, the sent message carries a wrong checksum, the client asks to
-        execute an unknown command.
-        Additional information here:
+        """This method receives and parses the command to be sent to the
+        System. Additional information here:
         https://github.com/discos/simulators/issues/1
 
-        :param byte: the received message byte."""
+        :param byte: the received message byte.
+        :type byte: byte
+        :return: False when the given byte is not the header, but the header is
+            expected. True when the given byte is the header or a following
+            expected byte. The response (the string to be sent back to the
+            client) when the message is completed.
+        :rtype: boolean, string
+        :raise ValueError: when the declared length of the message exceeds the
+            maximum expected length, when the sent message carries a wrong
+            checksum or when the client asks to execute an unknown command."""
 
 
 class SendingSystem(BaseSystem):
+    """This class implements a server that, as soon as a client(s) opens a
+    connection, it starts periodically sending to the said client(s)
```

Comment:
connection, starts (without it). Actually the sentence is not clear to me. Maybe: *Implement a server that sends some system information data to every connected client.*